addons/project/models/project_task.py

```diff
@@ -992,28 +985,38 @@ def _ensure_fields_are_accessible(self, fields, operation='read', check_group_us
         """
         assert operation in ('read', 'write'), 'Invalid operation'
         if fields and (not check_group_user or self.env.user._is_portal()) and not self.env.su:
-            unauthorized_fields = set(fields) - (self.SELF_READABLE_FIELDS if operation == 'read' else self.SELF_WRITABLE_FIELDS)
+            readable, writeable = self._portal_accessible_fields()
+            unauthorized_fields = set(fields) - (readable if operation == 'read' else writeable)
             if unauthorized_fields:
                 if operation == 'read':
                     error_message = _('You cannot read the following fields on tasks: %(field_list)s', field_list=unauthorized_fields)
                 else:
                     error_message = _('You cannot write on the following fields on tasks: %(field_list)s', field_list=unauthorized_fields)
                 raise AccessError(error_message)
 
+    @api.model
+    @tools.ormcache()
+    def _portal_accessible_fields(self) -> tuple[frozenset[str], frozenset[str]]:
+        """Readable and writable fields by portal users."""
+        readable = frozenset(self.SELF_READABLE_FIELDS)
+        writeable = frozenset(self.SELF_WRITABLE_FIELDS)
+        return readable | writeable, writeable
```

Comment:
TBH, I am not sure to understand why it is needed to cache the both sets. Why not directly using the both getters?

Moreover, `SELF_READABLE_FIELDS` already contains the writable fields. 🙂

Reply:
Because it is used in `_has_field_access` which is performance-critical for the ORM.
Existing properties are overwritten and with each overwrite there is a function call and a new list rebuilt.

All writeable fields are readable here, so it is just forced by making a union between the sets.